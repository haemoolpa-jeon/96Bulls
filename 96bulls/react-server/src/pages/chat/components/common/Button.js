import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.scss';

class Button extends Component {
  handleClick = (e) => {
    const { onClick } = this.props;
    onClick && onClick(e);
  };

  render() {
    const {
      children, loader, disabled, className,
    } = this.props;
    return (
      <button
        className={`${styles.button} ${className}`}
        onClick={this.handleClick}
        type="button"
        disabled={disabled}
      >
        { loader ? <div className={styles.loader} /> : children }
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  loader: PropTypes.bool,
  disabled: PropTypes.bool,
  className: PropTypes.string,
};

Button.default = {
  loader: false,
  disabled: false,
};

export default Button;
