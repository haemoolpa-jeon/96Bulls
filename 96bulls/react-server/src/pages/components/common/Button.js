/* Frequently used Button Component */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.scss';

class Button extends Component {
  handleClick = (e) => {
    const { onClick } = this.props;
    onClick && onClick(e);
  };

  render() {
    const {
      className, disabled, children, loader,  
    } = this.props;
    return (
      <button
        className={`${styles.button} ${className}`}
        disabled={disabled}
        onClick={this.handleClick}
        type="button"
      >
        { loader ? <div className={styles.loader} /> : children }
      </button>
    );
  }
}

Button.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.node.isRequired,
  loader: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.default = {
  loader: false,
  disabled: false,
};

export default Button;
