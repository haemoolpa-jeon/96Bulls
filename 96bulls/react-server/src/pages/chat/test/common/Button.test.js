import React from 'react';
import Button from '../../components/common/Button';
import {shallow} from 'enzyme/build';

describe('Button', () => {
  let component = null;
  const mockCallback = jest.fn(e => e);
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<Button
      onClick={mockCallback}>
      버튼
    </Button>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });

  it('onClick 함수가 수행되어야 함', () => {
    component.find('button').simulate('click');
    expect(mockCallback.mock.calls.length).toBe(1);
  });

  it('div.loader를 랜더링 해야함', () => {
    component = shallow(<Button
      loader
      onClick={mockCallback}>
      버튼
    </Button>);
    component.find('div.loader').exists();
  });
});
