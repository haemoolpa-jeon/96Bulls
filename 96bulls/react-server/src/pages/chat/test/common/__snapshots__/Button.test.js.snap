// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 초기 렌더링 스냅샷 일치함 1`] = `
<button
  className="button undefined"
  onClick={[Function]}
  type="button"
>
  버튼
</button>
`;
